ovh/resource_cloud_project_gateway.go

```diff
@@ -0,0 +1,265 @@
+package ovh
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/ovh/terraform-provider-ovh/ovh/helpers"
+
+	"github.com/ovh/go-ovh/ovh"
+)
+
+func resourceOvhCloudProjectGatewayImportState(
+	d *schema.ResourceData,
+	meta interface{},
+) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	splitId := strings.SplitN(givenId, "/", 2)
+	if len(splitId) != 2 {
+		return nil, fmt.Errorf("Import Id is not OVH_CLOUD_PROJECT/network_id formatted")
+	}
+	d.SetId(splitId[1])
+	d.Set("service_name", splitId[0])
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
+}
+
+func resourceCloudProjectGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudProjectGatewayCreate,
+		Read:   resourceCloudProjectGatewayRead,
+		Update: resourceCloudProjectGatewayUpdate,
+		Delete: resourceCloudProjectGatewayDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceOvhCloudProjectGatewayImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"service_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OVH_CLOUD_PROJECT_SERVICE", nil),
+				Description: "Service name of the resource representing the id of the cloud project.",
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"model": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"network_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudProjectGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	serviceName := d.Get("service_name").(string)
+	region := d.Get("region").(string)
+	network := d.Get("network_id").(string)
+	subnet := d.Get("subnet_id").(string)
+
+	params := &CloudProjectGatewayCreateOpts{
+		Name:  d.Get("name").(string),
+		Model: d.Get("model").(string),
+	}
+
+	r := &CloudProjectGatewayResponse{}
+
+	log.Printf("[DEBUG] Will create public cloud gateway: %s", params)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/region/%s/network/%s/subnet/%s/gateway", serviceName, region, network, subnet)
```

Comment:
- [ ] you should also handle the basic use case where the subnet is created alongside the gateway using `POST /cloud/project/{serviceName}/region/{regionName}/gateway`, and update the schema accordingly.

Reply:
like i said , the description i've just split the previous MR, because, there were no test and documentation about it , i did not cahnge the code base..

can you please, share me a shunk of code doing quite the same thing ? ( first time doing TF stuff )

Reply:
OK I see, it could be acceptable to handle only the case when you already have a subnet and just want to create a gateway, but it should be clearly mentioned at the beginning of the resource's doc.

> can you please, share me a chunk of code doing quite the same thing ?

I don't have ready-to-use code for this, but I think that it wouldn't take too much time to implement both use cases (with/without a pre-provisionned subnet). You could:
- set property `subnet_id`optional
- add the properties needed for `POST /cloud/project/{serviceName}/region/{regionName}/gateway` (optional also)
- in the create func, if subset_id is nil, use route `POST /cloud/project/{serviceName}/region/{regionName}/gateway`, else use your current API call

Reply:
thank you for the answer , will add some info in the doc about it .

like i said i don't have too much time , to implement all of that . will try to spend more time asap.

but meanwhile i've fixed all the remarks 

thank you =)